src/resolver/resolver.zig

```diff
@@ -1930,20 +1937,22 @@ pub const Resolver = struct {
 
     fn handleESMResolution(r: *ThisResolver, esm_resolution_: ESModule.Resolution, abs_package_path: string, kind: ast.ImportKind, package_json: *PackageJSON) ?MatchResult {
         var esm_resolution = esm_resolution_;
-        if (!((esm_resolution.status == .Inexact or esm_resolution.status == .Exact) and
+        if (!((esm_resolution.status == .Inexact or esm_resolution.status == .Exact or esm_resolution.status == .ExactEndsWithStar) and
             esm_resolution.path.len > 0 and esm_resolution.path[0] == '/'))
             return null;
 
         const abs_esm_path: string = brk: {
             var parts = [_]string{
                 abs_package_path,
-                esm_resolution.path[1..],
+                esm_resolution.path,
```

Comment:
I think `esm_resolution.path` often starts with `/`. Joining filesystem paths that start with `/` become an absolute path. This would cause `foo/bar/baz.js` to be `/foo/bar/baz.js` instead of `${pwd}/node_modules/foo/bar/baz.js`

Instead, maybe it should be: 

```suggestion
                strings.withoutLeadingSlash(esm_resolution.path),
```

Reply:
I guess you're right, I was basing it on this change: https://github.com/evanw/esbuild/commit/ccd8cd45a5a56fdff35bb035438d17a005f3cf96#diff-ff3593e6a6cb00f927118c829760e054ccca7686eae492f9f4f18b9b851a9a20L2137
But I guess Go's `Path.join()` behaves differently here.

Reply:
Ok, I updated it based on your comment